File: abstract_classes/abstract_classes_java/AbstractQueue.java
package abstract_classes.abstract_classes_java;

public abstract class AbstractQueue {
    protected int size;

    public AbstractQueue() {
        size = 0; 
    }

    public abstract void enqueue(int value); 

    public abstract int dequeue(); 

    public int size() {
        return size;
    }

    public boolean isEmpty() {
        return size == 0;
    }S
}

File: abstract_classes/abstract_classes_c++/AbstractQueue.cpp
namespace abstract_classes {
    class AbstractQueue {
    protected:
        int size;

    public:
        AbstractQueue() : size(0) {}

        virtual void enqueue(int value) = 0;
        virtual int dequeue() = 0;

        int getSize() {
            return size;
        }

        bool isEmpty() {
            return size == 0;
        }
    };
}